src/OpenTelemetry.AutoInstrumentation.Native/logger_impl.h

```diff
@@ -131,28 +138,41 @@ LoggerImpl<TLoggerPolicy>::LoggerImpl()
 
     spdlog::flush_every(std::chrono::seconds(3));
 
-    static auto current_process_name = ToString(GetCurrentProcessName());
-    static auto current_process_id   = GetPID();
-    static auto current_process_without_extension =
-        current_process_name.substr(0, current_process_name.find_last_of("."));
+    static auto configured_log_sink = GetLogSinkConfiguration();
+
+    if (configured_log_sink == log_sink_none)
+    {
+        m_fileout = spdlog::null_logger_mt(logger_name);
+    }
+    else if (configured_log_sink == log_sink_file)
+    {
+        static auto current_process_name = ToString(GetCurrentProcessName());
+        static auto current_process_id   = GetPID();
+        static auto current_process_without_extension =
+            current_process_name.substr(0, current_process_name.find_last_of("."));
 
-    static auto file_name_suffix =
-        std::to_string(current_process_id) + "-" + current_process_without_extension + "-Native";
+        static auto file_name_suffix =
+            std::to_string(current_process_id) + "-" + current_process_without_extension + "-Native";
 
-    // by default, use the same size as on managed side: 10MiB
-    static auto file_size        = GetConfiguredSize(environment::max_log_file_size, 10485760);
+        // by default, use the same size as on managed side: 10MiB
+        static auto file_size = GetConfiguredSize(environment::max_log_file_size, 10485760);
 
-    try
-    {
-        m_fileout = spdlog::rotating_logger_mt(logger_name, GetLogPath(file_name_suffix), file_size, 10);
+        try
+        {
+            m_fileout = spdlog::rotating_logger_mt(logger_name, GetLogPath(file_name_suffix), file_size, 10);
+        }
+        catch (...)
+        {
+            // By writing into the stderr was changing the behavior in a CI scenario.
+            // There's not a good way to report errors when trying to create the log file.
+            // But we never should be changing the normal behavior of an app.
+            // std::cerr << "LoggerImpl Handler: Error creating native log file." << std::endl;
+            m_fileout = spdlog::null_logger_mt(logger_name);
+        }
```

Comment:
Now that this is inside a conditional block it seems that it deserves a separate function, or at least to be last branch of the if statement.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3558/commits/b434bd17e4b10337d1142c5b6e008f15bafa87a8